### ISSUE TITLE
FSI throws FileLoadException at startup from within VS2017

### ISSUE THREAD

**Issue by chrnola** [Microsoft/visualfsharp#4390]
#### Repro steps

1. Upgrade to Visual F# Tools VSIX version 15.4.1.18022303

2. Try to launch F# Interactive from Visual Studio 2017 15.5.7 and observe the following exception:
```
Microsoft (R) F# Interactive version 4.1
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> 

error FS0193: Could not load file or assembly 'System.ValueTuple, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Exception by fsi.exe:
System.Exception: Error creating evaluation session: System.IO.FileLoadException: Could not load file or assembly 'System.ValueTuple, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'System.ValueTuple, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
   at Microsoft.FSharp.Compiler.CompileOps.GetDefaultSystemValueTupleReference()
   at Microsoft.FSharp.Compiler.CompileOps.DefaultReferencesForScriptsAndOutOfProjectSources@1809.GenerateNext(IEnumerable`1& next)
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl()
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System-Collections-IEnumerator-MoveNext()
   at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source)
   at Microsoft.FSharp.Compiler.CompileOps.GetAllDllReferences@3692.GenerateNext(IEnumerable`1& next)
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl()
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System-Collections-IEnumerator-MoveNext()
   at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source)
   at Microsoft.FSharp.Compiler.CompileOps.TcAssemblyResolutions.SplitNonFoundationalResolutions(CompilationThreadToken ctok, TcConfig tcConfig)
   at <StartupCode$FSharp-Compiler-Private>.$IncrementalBuild.Get@1055.Invoke(Unit unitVar)
   at Microsoft.FSharp.Compiler.AbstractIL.Internal.Library.CancellableModule.delay@751.Invoke(CancellationToken ct)
   at Microsoft.FSharp.Compiler.AbstractIL.Internal.Library.CancellableModule.runWithoutCancellation[a](Cancellable`1 comp)
   at Microsoft.FSharp.Compiler.Interactive.Shell.FsiEvaluationSession..ctor(FsiEvaluationSessionHostConfig fsi, String[] argv, TextReader inReader, TextWriter outWriter, TextWriter errorWriter, Boolean fsiCollectible, FSharpOption`1 legacyReferenceResolver)

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

   at Microsoft.FSharp.Core.PrintfModule.PrintFormatToStringThenFail@1645.Invoke(String message)
   at Microsoft.FSharp.Compiler.Interactive.Shell.FsiEvaluationSession..ctor(FsiEvaluationSessionHostConfig fsi, String[] argv, TextReader inReader, TextWriter outWriter, TextWriter errorWriter, Boolean fsiCollectible, FSharpOption`1 legacyReferenceResolver)
   at Microsoft.FSharp.Compiler.Interactive.Shell.FsiEvaluationSession.Create(FsiEvaluationSessionHostConfig fsiConfig, String[] argv, TextReader inReader, TextWriter outWriter, TextWriter errorWriter, FSharpOption`1 collectible, FSharpOption`1 legacyReferenceResolver)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Sample.Microsoft.FSharp.Compiler.Interactive.Main.evaluateSession(String[] argv)
```

#### Expected behavior

FSI starts without throwing an exception.

#### Actual behavior

See above. FSI throws an exception and does not accept any user input.

#### Known workarounds

* Downgrade to VSIX 15.4.1.18022203

#### Related information 

* Windows 10
* Nightly VSIX (15.4.1.18022303)
* Full .NET Framework 4.6.2
* Visual Studio 2017 Enterprise 15.5.7

**Comment by dsyme** [comment]
@cartermp @KevinRansom Can you repro this? thanks

**Comment by KevinRansom** [comment]
Seems to work for me:

![image](https://user-images.githubusercontent.com/5175830/36697885-261e8f48-1afd-11e8-8547-a5601a5e28bd.png)

**Comment by dsyme** [comment]
@chrnola Which .NET Framework do you have installed on your machine?

**Comment by cartermp** [comment]
From the issue:

> .NET Framework 4.7.1

Which shouldn't be a problem. @chrnola can you reproduce this with the latest nightly?

**Comment by chrnola** [comment]
My apologies, the issue is incorrect. I have .NET Framework 4.6.2 installed (more specifically, the value of `HKLM\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full` is 394806).

But yes, I can still reproduce the issue with the latest nightly. Currently running version 15.4.1.18022703 of the VSIX.

**Comment by cartermp** [comment]
`System.ValueTuple` is not available in .NET Framework 4.6.2, so that explains the file load error. However, I'm not sure why this would cause F# Interactive to fail with a straight VS installation of 15.5.7. @KevinRansom any ideas?

**Comment by dsyme** [comment]
> System.ValueTuple is not available in .NET Framework 4.6.2, so that explains the file load error. However, I'm not sure why this would cause F# Interactive to fail with a straight VS installation of 15.5.7. 

@cartermp `fsi.exe` (actually FSharp.Compiler.Private.dll) should depend on and find the System.ValueTuple.dll we ship alongside it in this case.  But something is screwed up - we've somehow picked up a dependency on the wrong System.ValueTuple.dll. 

Anyway, it sounds like a fairly bad bug.

**Comment by cartermp** [comment]
@chrnola If you uninstall the nightly, does this repro? I suspect that you may have picked up an upgraded version via the nightly feed on master that doesn't reflect what is actually released in 15.5. I wouldn't advise using nightlies anymore unless you want to be on the bleeding edge.  Uninstalling the nightly and upgrading to 15.6 will give you all the updated versions for everything.

@dsyme If this doesn't repro on a regular release then I don't think it's a bad bug; just another one of the nightly release "sync" issues.

**Comment by dsyme** [comment]
@chrnola Please reopen if this repro's with the nightlies uninstalled
